addons/purchase_stock/models/stock_rule.py

```diff
@@ -48,11 +48,15 @@ def _run_buy(self, procurements):
             procurement_date_planned = fields.Datetime.from_string(procurement.values['date_planned'])
             schedule_date = (procurement_date_planned - relativedelta(days=procurement.company_id.po_lead))
 
-            supplier = procurement.product_id._select_seller(
-                partner_id=procurement.values.get("supplier_id"),
-                quantity=procurement.product_qty,
-                date=schedule_date.date(),
-                uom_id=procurement.product_uom)
+            supplier = False
+            if procurement.values.get('supplier_id') and procurement.values['supplier_id']._name == 'product.supplierinfo':
+                supplier = procurement.values['supplier_id']
+            else:
+                supplier = procurement.product_id._select_seller(
+                    partner_id=procurement.values.get("supplier_id"),
```

Comment:
can you receive a supplier_id which isn't a supplier infO?

Reply:
Before my patch _select_seller use partner_id=procurement.values.get("supplier_id")
In this cast supplier_id is a partner from values. We don't use it. It comes from the patch that allow to use a partner in values but we could modify the dict key also